clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/customtemplates/CustomTemplateContext.kt

```diff
@@ -238,9 +280,20 @@ sealed class CustomTemplateContext private constructor(
      * See [CustomTemplateContext].
      */
     class FunctionContext internal constructor(
-        template: CustomTemplate,
-        notification: CTInAppNotification,
-        inAppListener: InAppListener,
-        logger: Logger
-    ) : CustomTemplateContext(template, notification, inAppListener, logger)
+        template: CustomTemplate, notification: CTInAppNotification, inAppListener: InAppListener, logger: Logger
+    ) : CustomTemplateContext(template, notification, inAppListener, logger) {
+
+        private val isVisual = template.isVisual
+
+        /**
+         * Visual functions ([CustomTemplate.isVisual]` = true`) are considered as a message that is visible to the
+         * user. See [TemplateContext.setDismissed] for more details. Non-visual functions are executed directly and
+         * do not require to be explicitly dismissed.
+         */
+        override fun setDismissed() {
+            if (isVisual) {
```

Comment:
`if (isVisual)` check can be moved in `CustomTemplateContext` itself?

Reply:
Yes, I have moved it in another commit in a subsequent PR.